A36-xds-for-servers.md

```diff
@@ -0,0 +1,401 @@
+trivial code change to opt-in to xDS, and from that point on newly-added
+xDS-powered gRPC features can be used without code changes.
+
+### Related Proposals:
+
+ * [A27: xDS-Based Global Load Balancing][A27]
+ * [A29: xDS-Based Security for gRPC Clients and Servers][A29]
+
+[A27]: A27-xds-global-load-balancing.md
+[A29]: A29-xds-tls-security.md
+
+## Proposal
+
+Each language will create an "XdsServer" API that will wrap the normal "Server"
+API. This will allow the implementation to see the listening ports, start event,
+stop event, and also allow it access to the server to change behavior (e.g., via
+an interceptor). Users will use the XdsServer API to construct the server
+instead of the existing API. XdsServer will not support ephemeral ports as part
+of this gRFC but may be enhanced in the future.
+
+Credential configuration will be managed separately by providing
+XdsServerCredentials, similar to client-side. The user must pass the
+XdsServerCredentials to the XdsServer API to enable control-plane managed TLS
+certificates. While the user-facing API is discussed here, the precise
+implementation details are covered in gRFC A29. The user is free to pass other
+ServerCredentials types to XdsServer and they will be honored. Note that RBAC is
+authz, not authn, so it would be enabled by using the XdsServer, even without
+XdsServerCredentials.
+
+### Serving and Not Serving
+
+To serve RPCs, the XdsServer must have its xDS configuration, provided via a
+Listener resource and potentialy other related resources. When its xDS
+configuration is missing or unknown the server must be in a "not serving" mode.
+This is ideally implemented by not `listen()`ing on the port. If that is
+impractical an implementation may be `listen()`ing on the port, but it must also
+`accept()` and immediately `close()` connections, making sure to not send any
+data to the client (e.g., no TLS ServerHello nor HTTP/2 SETTINGS). With either
+behavior, client connection attempts will quickly fail and clients would not
+perform further attempts without a backoff. Load balancing policies like
+`pick_first` would naturally attempt connections to any remaining addresses to
+quickly find an operational backend. However, the `accept()`+`listen()` approach
+will be improperly detected as server liveness for TCP heath checking.
+
+If the xDS bootstrap is missing or invalid, implementations would ideally fail
+XdsServer startup, but it is also acceptable to consider it a lack of xDS
+configuration and enter a permanent "not serving" mode.
+
+If the server is unable to open serving ports (e.g., because the port is already
+in use by another process), XdsServer startup may fail or it may enter "not
+serving" mode. If entering "not serving" mode, opening the port must be retried
+automatically (e.g., retry every minute).
+
+Communication failures do not impact the XdsServer's xDS configuration; the
+XdsServer should continue using the most recent configuration until connectivity
```

Comment:
Is that true also for the very first time config is needed during `start()'? What's the expected behavior when the listener resource is missing at startup time? Should the server's `start()` (in Java) fail with the server going into permanent failure state?